### DISCUSSION TITLE
Issue with iterating annotations

### DISCUSSION THREAD

**Discussion by mwade625** [pymupdf/PyMuPDF#2230] [Q&A]
I have a loop to iterate the annotations on a given page and then clear them using a loop as shown below.  This has always worked for me, but I have encountered a situation where the load_annot() in the fitz library throws the following exception:

"runtimeerror: xref 175 is not an annot of this page"

```
   for page in doc:
      for annot in page.annots():
         page.delete_annot(annot)
         
      # Added this to see if this would resolve the issue as it is recommended to call this if we are doing anything 
      # with annotations on this page after the above call.
      page = doc.reload_page( page )``
```

I then modified this loop to work as follows:

```
   for page in doc:
      annot = page.firstAnnot
      while annot:
         annot = page.delete_annot(annot)
```

Is this expected behavior?  And what is the recommended way to do this?  Please note that I am also adding NEW annotations after the annotations are cleared for all pages.

Thanks

Mike

**Comment by JorjMcKie** [comment]
You did not mention your PyMuPDF version. In some version, under certain conditions, the first loop version did not ignore invalid entries in the page's `/Annots` array and tried to treat junk entries like an annotation.

Should be resolved now, so both loops should always work with the newest version.

**Comment by mwade625** [comment]
Thanks for the response kind response.  I apologize for that very poor problem description.  I am using the very latest version 1.21.1 and python 3.9.  I have switched over to the 2'nd style which works fine, but I was surprised at the exception.
